TKeazirian.HTTPServer/Server/Router.cs

```diff
@@ -19,9 +19,17 @@ public Response Route(Request request)
             IsHttpMethodAllowed(request))
         {
             var handler = _routesConfig.Routes[request.GetRequestPath()];
-            return request.GetRequestMethod() == "HEAD"
-                ? HeadRequest(handler, request)
-                : handler.HandleResponse(request);
+            if (request.GetRequestMethod() == "HEAD" && handler.AllowedHttpMethods().Contains("GET"))
```

Comment:
Re: https://github.com/t-keazirian/HTTPServer/pull/27#discussion_r911368942

Would this be considered enforcing that it has a GET request if you are asking for an OPTIONS or HEAD request?